redis/client.py

```diff
@@ -1144,11 +1149,11 @@ def _send_command_parse_response(self, conn, command_name, *args, **options):
     def _disconnect_raise(self, conn, error):
         """
         Close the connection and raise an exception
-        if retry_on_timeout is not set or the error
-        is not a TimeoutError
+        if retry_on_error is not set or the error
+        is not one of the specified error types
         """
         conn.disconnect()
-        if not (conn.retry_on_timeout and isinstance(error, TimeoutError)):
+        if not (conn.retry_on_error and isinstance(error, tuple(conn.retry_on_error))):
```

Comment:
What do you think about making this more expressive and positive cased? I'm hoping the below, makes this more clear to the next person who reads this.

```
if (conn.retry_on_error is None and isinstance(error, tuple(conn.retry_on_error))) is False:
    raise error
```

Reply:
changed to:
```
        if (
            conn.retry_on_error is None
            or isinstance(error, tuple(conn.retry_on_error)) is False
        ):
            raise error
```